src/utils/merkle_tree.zig

```diff
@@ -0,0 +1,125 @@
+const std = @import("std");
+const sig = @import("../sig.zig");
+const Hash = sig.core.Hash;
+const Sha256 = std.crypto.hash.sha2.Sha256;
+
+pub fn NestedList(comptime T: type) type {
+    return struct {
+        items: []const []T,
+        const Self = @This();
+
+        pub fn getValue(self: *const Self, index: u64) *T {
+            std.debug.assert(index < self.len());
+            var search_index: usize = 0;
+            for (self.items) |slice| {
+                if (search_index + slice.len > index) {
+                    const index_in_nested = index - search_index;
+                    return &slice[index_in_nested];
+                } else {
+                    search_index += slice.len;
+                }
+            }
+            unreachable;
+        }
+
+        pub fn getSlice(self: *const Self, index: u64, length: u64) []T {
```

Comment:
What exactly is the point of this function? I don't see it used anywhere so it's hard to judge what exactly the point is; there's no API for getting the length of the slice that the element at `index` would be in, which makes me question how exactly the `length` should be computed by the caller, or what the usage is in general.

Reply:
ah great catch - it was some old code i used previously but dont need anymore

d89bf4a